### ISSUE TITLE
Accessing capture components from within code when HUD sub views are disabled

### ISSUE THREAD

**Issue by saihv** [Microsoft/AirSim#194]
For better performance, I am experimenting how things work if the depth, seg and scene sub views are turned off within the AirSim HUD (the draw thread is much faster with these off). Upon first looks, it seems that the capture components within the code are returning NULL when these sub-views are turned off, which in turn affects camera recording etc. : is it possible to keep these views disabled and yet be able to access the capture components/images from inside the code?

**Comment by bretStaz** [comment]
The performance gains you are seeing are most likely not from turning off the rendering of the additional views, but instead the turning off of the generation of the data that makes up those views so even if you generate the data but do not display them you will probably see the same rough performance as you see when displaying them to the screen.   So this direction most likely will not yield performance gains you are looking for.

**Comment by bretStaz** [comment]
My current guess is the segmentation camera is the most expensive to calculate, I want to have depth camera views for all six sides ideally and not the segmentation.  I will probably start to work on that and let you know what the performance is like without the segmentation camera. 

The Camera Director in the Unreal plugin has a "bool togglePIPSeg();" function that does not look like it is exposed to the client side but exposing each camera toggle to the client would achieve let someone turn off the segmentation camera and keep the others.  

Depth data is probably expensive to calculate (unless they are using hardware ) which is why the viewport for the depth data is smaller then the visual data.

**Comment by bretStaz** [comment]
finally found your other thread about depth and segmentation being done via shaders so that answers a lot of my questions about performance impact on segmentation.  Have your improvement for multi-thread logging been merged into master, those changes are appealing.

**Comment by Dynode** [comment]
@bretStaz They're added.

**Comment by saihv** [comment]
@bretStaz  Thanks for the insight! Quick question: you mentioned that disabling the views would disable the generation of data entirely, but if I remember correctly, even when I pressed 0 to disable the sub-views, I was able to double click on the FpvRenderTarget (or any other render target) within Unreal and still be able to see the stream of camera video.  Have you ever tried something like this? I am going to check this again though because it was quite some time ago and I may be mistaken.
